### PR TITLE
[Feature] Add ActiveMQ as streaming data source

### PR DESCRIPTION
**Pull request by shrutimantri**
# Description
Added ActiveMQ as streaming data source
Screenshot 2023-11-14 at 6 18 26 PM

# How Has This Been Tested?
- [ Yes ] Added SourceFactory test case for ActiveMQ
- [ Yes ] Added ActiveMQ client initialization test case

# Checklist
- [ Yes ] The PR is tagged with proper labels (bug, enhancement, feature, documentation)
- [ Yes ] I have performed a self-review of my own code
- [ Yes ] I have added unit tests that prove my fix is effective or that my feature works
- [ Yes ] I have commented my code, particularly in hard-to-understand areas
- [ Yes ] I have made corresponding changes to the documentation
- [ Yes ] If new documentation has been added, relative paths have been added to the appropriate section of `docs/mint.json`

cc:

### CODE REVIEW

File: mage_ai/streaming/sources/activemq.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    connection_host: str
+    connection_port: int
+    queue_name: str
+    configure_consume: bool = False
+    username: str = 'admin'
+    password: str = 'admin'
+
+
+def messageProcessingFunction(message, handler):
+    print('Recieved message: "%s"' % message)
```
Comment (wangxiaoyou1993) [inline mage_ai/streaming/sources/activemq.py:23]: what are the possible types of the message? can it be bytes, str, dict? or is it just a string?

File: mage_ai/streaming/sources/activemq.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    handler(message)
```
Comment (wangxiaoyou1993) [inline mage_ai/streaming/sources/activemq.py:24]: Usually, the handler expects the input to be a list of messages: https://github.com/mage-ai/mage-ai/blob/master/mage_ai/streaming/sinks/base.py#L42

File: mage_ai/streaming/sources/activemq.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    def read(self, handler: Callable):
+        pass
+
+    def batch_read(self, handler: Callable):
+        self._print('Start consuming messages.')
+        listener = ActiveMQMsgListener(processMessage=messageProcessingFunction,
+                                       conn=self.connection,
+                                       handler=handler)
+        self.connection.set_listener('ActiceMQListerner', listener)
+        self.connection.subscribe(destination=f'/queue/{self.config.queue_name}', id=1, headers={})
```
[the remark this answers is not in the document]
Comment (shrutimantri) [reply mage_ai/streaming/sources/activemq.py:79]: This needs to be unique. Changed it to `uuid.uuid4()`.

File: mage_ai/streaming/sources/activemq.py (hunk 2 above)
Comment (shrutimantri) [reply mage_ai/streaming/sources/activemq.py:24]: Its not very clear from the stomp library whether **batch_read** is even possible. But given that ActiveMQ is not a designed to be a distributed system, it won't have extreme heavy loads, and hence, it should be ok to have handler called for each message.

File: mage_integrations/mage_integrations/tests/sources/api/test_api.py [hunk 4]
```diff
@@ -216,6 +216,8 @@ def json_catalog_example():
 
 
 class ApiTest(unittest.TestCase):
+    maxDiff = None
```
Comment (wangxiaoyou1993) [inline mage_integrations/mage_integrations/tests/sources/api/test_api.py:219]: what's this for?

Comment (shrutimantri) [reply mage_integrations/mage_integrations/tests/sources/api/test_api.py:219]: It was for some unrelated test failures. The test failure was not giving complete diff. Removing this.

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**